src/utils/generate_cxx_code.cpp

```diff
@@ -759,39 +1048,346 @@ const std::string  wcs::generate_cxx_code::generate_code(const LIBSBML_CPP_NAMES
+        var_names.insert(*it);
+      }
+    }
+    std::string formula = SBML_formulaToString(reaction.getKineticLaw()->getMath());
+    const auto input_map = build_input_map(formula, var_names);
+
+    std::vector<std::string> var_names_ord (input_map.size());
+    for (auto& x: input_map) {
+      var_names_ord[x.second] = x.first;
+    }
+
+    //print first parameters in formula taking input
+    std::vector<std::string>::iterator it;
+    std::vector<std::string> par_names, par_names_nf;
+    all_var_names = var_names;
+    int par_index = 0;
+    for (it = var_names_ord.begin(); it < var_names_ord.end(); it++){
+      rrdit = rate_rules_dep_map.find(*it);
+      evassigit = m_ev_assig.find(*it);
+      if (rrdit != rate_rules_dep_map.cend()) { //rate rules
+        std::set<std::string> params_fn = rrdit->second;
+        std::set<std::string> params_fn_dec;
+        std::string function_input;
+        std::set<std::string>::iterator itf;
+        // add all the dependent species of rate rules
+        for (itf = params_fn.begin(); itf != params_fn.end(); itf++){
+          if (itf != params_fn.begin()) {
+           function_input = function_input + ", ";
+          }
+          var_names_it = var_names.find(*itf);
+          all_var_names_it = all_var_names.find(*itf);
+          if (all_var_names_it == all_var_names.cend()) {
+            genfile << "  " << Real << " " << *itf <<" = __input[" << par_index++ << "];\n";
+            var_names.erase(*itf);
+          } else {
+            if (var_names_it != var_names.cend()) {
+              genfile << "  " << Real << " " << *itf <<" = __input[" << par_index++ << "];\n";
+              var_names.erase(*itf);
+            }
+          }
+          function_input = function_input + *itf ;
+        }
+        genfile << "  wcs_global_var."  << *it << " = wcs__rate_" << *it << "(std::vector<" << Real << "> {" << function_input << "});\n";
+        par_names.push_back(*it);
+      } else if (evassigit != m_ev_assig.cend()) { //events variables
+        std::string function_input;
+        function_input ="";
+        genfile << "  wcs_global_var." << *it << " = wcs__rate_" << *it << "(" << function_input << ");\n";
+      } else {
+        genfile << "  " << Real << " " << *it <<" = __input[" << par_index++ << "];\n";
+        par_names.push_back(*it);
+      }
+      var_names.erase(*it);
+    }
+    //print rest parameters (not in formula) taking input
+    for (auto& x: var_names) {
+      rrdit = rate_rules_dep_map.find(x);
+      evassigit = m_ev_assig.find(x);
+      if (rrdit != rate_rules_dep_map.cend()) { //rate rules
+        std::set<std::string> params_fn = rrdit->second;
+        std::string function_input ;
+        std::set<std::string>::iterator itf;
+        // add all the dependent species of rate rules
+        for (itf = params_fn.begin(); itf != params_fn.end(); itf++){
+          if (itf != params_fn.begin()) {
+           function_input = function_input + ", ";
```

Comment:
indent 2 spaces
same goes for the lines below

Reply:
Done